packages/f/ffmpeg/xmake.lua

```diff
@@ -45,6 +45,50 @@ package("ffmpeg")
         add_deps("yasm")
     end
 
+    if is_plat("mingw") and is_subhost("msys") then
+        add_extsources("pacman::ffmpeg")
+    elseif is_plat("linux") then
+        add_extsources("pacman::ffmpeg")
+    elseif is_plat("macosx") then
+        add_extsources("brew::ffmpeg")
+    end
+    
+    if on_fetch then
+        on_fetch("linux", function (package, opt)
+            if opt.system then
+                local result
+                for _, name in ipairs({"libavcodec", "libavdevice", "libavfilter", "libavformat", "libavutil", "libpostproc", "libswresample", "libswscale"}) do
+                    local pkginfo = package.find_package and package:find_package("pkgconfig::" .. name, opt)
+                    if pkginfo then
+                        if not result then
+                            result = table.copy(pkginfo)
+                        else
+                            local includedirs = pkginfo.sysincludedirs or pkginfo.includedirs
+                            result.links = table.wrap(result.links)
+                            result.linkdirs = table.wrap(result.linkdirs)
+                            result.includedirs = table.wrap(result.includedirs)
+                            table.join2(result.includedirs, includedirs)
+                            table.join2(result.linkdirs, pkginfo.linkdirs)
+                            table.join2(result.links, pkginfo.links)
+                        end
+                    end
+                end
+                result.includedirs = table.unique(result.includedirs)
+                result.linkdirs = table.unique(result.linkdirs)
+                import("lib.detect.find_programver")
+                
+                local version = find_programver("ffmpeg", {command = "-version"})
+                if(version == nil) then
+                    return
+                end
```

Comment:
we should use find_tool instead of find_programver

Reply:
find_tool does not work...

Reply:
How can I override the version command like in `find_programver`  with `find_tool`?

Reply:
```lua
 find_tool("ffmpeg", {check = "-help", version = true, command = "-version", parse = "%d+%.?%d+%.?%d+", force = true})
```

Reply:
```lua

    on_fetch("linux", function (package, opt)
        import("lib.detect.find_tool")
        if opt.system then
            local result
            for _, name in ipairs({"libavcodec", "libavdevice", "libavfilter", "libavformat", "libavutil", "libpostproc", "libswresample", "libswscale"}) do
                local pkginfo = package:find_package("pkgconfig::" .. name, opt)
                if pkginfo then
                    pkginfo.version = nil
                    if not result then
                        result = pkginfo
                    else
                        result = result .. pkginfo
                    end
                end
            end
            if result then
                local ffmpeg = find_tool("ffmpeg", {check = "-help", version = true, command = "-version", parse = "%d+%.?%d+%.?%d+", force = true})
                if ffmpeg then
                    result.version = ffmpeg.version
                end
            end
            return result
        end
    end)
```

```console
{ 
  version = "4.3.2",
  sysincludedirs = { 
    "/usr/include/x86_64-linux-gnu" 
  },
  links = { 
    "avcodec",
    "avformat",
    "avutil",
    "swresample" 
  },
  linkdirs = { 
    "/usr/lib/x86_64-linux-gnu" 
  } 
}
```

Reply:
Done. I switched to using `on_fetch` for every platform to have the right ffmpeg version.

I also return nothing if one of the lib is not installed.